### PR TITLE
Added feature: Google Drive backup

### PR DESCRIPTION
**Pull request by arr00**
I was going to contact you before doing this, but you didn't have an email on your profile. My phone restored itself at WWDC after updating to iOS 12 and my reading list was one of the only things I lost. This small feature will add a seamless feature to the app for google drive backups.

### CODE REVIEW

**Comment by AndrewBennet** [comment]
Hi there - thanks for the pull request! This is in fact the first pull request anyone (other than myself) has raised for this project 🎉

I’ve had a look at your message and changes, and I have a few comments:

### Losing your data
Sorry you lost your data! This is surprising and concerning. I believed that the direction where the database is stored is included in iCloud backups. This makes me uneasy; I will look into this in due course.

### Feature discussions
You make a good point when you mention that you couldn't find a way to email me before beginning this. I should make it easier to raise potential features for discussion. I'll look into adding some guidelines on contributions and some forum for their discussion (perhaps GitHub issues, or perhaps on a Wiki page.)

### Syncing
Adding remote sync has been part of my plan for the app for a while - it has been one of the most requested features. I've actually already started work on a sync-mechanism, using iCloud. You can see my progress on the `icloud` branch. I favour this over syncing to a Google Drive file for a few reasons:

- It will work without requiring any app-specific login
- It will allow changes to propagate between devices
- Allows differences to be synced: only modifications need to be uploaded

The last point especially makes me hesitate with this approach: if a user has many many books, the CSV file could be quite big. Uploading it every time the app goes to the background is overkill. On the `icloud` branch, I've added a bitmask to each book which tracks any changes properties, and the CloudKit framework lets me upload just the modified properties, when needed.

In conclusion, this feature doesn't quite align with my plan for the app right now - I hope to release a version with iCloud sync fairly soon, and I think it will be a bit confusing to have a Google Drive sync also.

Thanks again for your contribution - I look forward to hearing your thoughts.

**Comment by arr00** [comment]
**Backup**
My phone wasn't being backed up to iCloud so no need to research an issue there... That was my fault.

**Pull Request**
I completely understand your concerns with my implementation and appreciate the feedback. I'm looking forward to iCloud sync, and maybe some day in the future I'll have a usefull pull request. 

Thanks for making such a great app!

**Comment by AndrewBennet** [comment]
Thanks for clarifying the iCloud backup concern 👍 

I would absolutely welcome any future contributions. I'll work out a forum for feature request discussion in due course. Thanks again.
